lms/static/scripts/frontend_apps/components/BasicLTILaunchApp.js

```diff
@@ -300,6 +301,7 @@ export default function BasicLTILaunchApp() {
         students={grading.students}
         courseName={grading.courseName}
         assignmentName={grading.assignmentName}
+        onError={handleError}
```

Comment:
Because it now gets passed on to `LMSGrader` as an error handling callback, I wondered if I ought to wrap `handleError` in `useCallback`, but it takes so many arguments that I didn't think there would be much/any optimization and it was also complicated to wrap and type. I'm open to any and all feedback.

Reply:
A higher level issue though is whether it makes sense to pass the entire `handleError` callback in, which has a big API surface area, thus making the interface between BasicLTILaunchApp and LMSGrader more complex.

As I see it, you have a couple of options:

1. Continue to handle authorization errors that occur in LMSGrader in the parent BasicLTILaunchApp component, but pass down a simpler interface, eg. an `onAuthError` callback. The `BasicLTILaunchApp` component would then map this to a `handleError` call.
2. Handle re-authorization within the LMSGrader component itself. This would avoid reloading the assignment content. There is an `AuthButton` component which is used in the group set selection UI that can help with this.

Reply:
This is good feedback. I will give it a proverbial think.

Reply:
We're punting on further error-handling for now.